R/ppc-rootograms.R

```diff
@@ -45,28 +47,31 @@ ppc_rootogram <- function(y, yrep, style = c("standing", "suspended"),
 
   # prepare a table for y
   ty <- table(y)
-  ty <- sqrt(as.numeric(ty[match(x, rownames(ty))]))
-  if (suspended) {
-    ty <- ty - tyexp
+  ty <- sqrt(as.numeric(ty[match(xpos, rownames(ty))]))
+  if (style == "suspended") {
+    ty <- tyexp - ty
   }
   ty[is.na(ty)] <- 0
+  ypos <- ty / 2
+  if (style == "hanging")
+    ypos <- tyexp - ypos
 
-  data <- data.frame(x, ty, tyexp, tyquantile)
+  data <- data.frame(xpos, ypos, ty, tyexp, tyquantile)
   graph <- ggplot(data) +
-    aes_(ymin = ~ tylower, ymax = ~ tyupper) +
-    geom_col(
-      aes_(x = ~ x, y = ~ ty, fill = "Observed"),
+    aes_(ymin = ~ tylower, ymax = ~ tyupper, height = ~ ty) +
```

Comment:
Non, I believe. The thing is that `ymin` and `ymax` are used by `geom_smooth` but can currently not be specified within it (this will be fixed in the next release of **ggplot2**). Same with `height` for `geom_tile`.